modin/numpy/math.py

```diff
@@ -44,10 +44,9 @@ def absolute(
 def add(
     x1, x2, out=None, where=True, casting="same_kind", order="K", dtype=None, subok=True
 ):
+    x1 = try_convert_from_interoperable_type(x1)
     if not isinstance(x1, array):
-        ErrorMessage.single_warning(
-            f"Modin NumPy only supports objects of modin.numpy.array types for add, not {type(x1)}. Defaulting to NumPy."
-        )
+        ErrorMessage.bad_type_for_numpy_op("add", type(x1))
```

Comment:
why don't we convert `x2` the same way? here and for other binary functions

Reply:
We don't need to - since the `binary_op` code takes care of converting x2!